From e9ffd084ec1ff9f7bfc86879732953dc58256958 Mon Sep 17 00:00:00 2001
From: Loko Kung <lokokung@google.com>
Date: Tue, 3 May 2022 00:28:53 +0000
Subject: [PATCH] Wrap get_gitHash in try-catch to prevent failures in tarball
 builds.

Bug: chromium:1321370
Change-Id: If39d2236d1b4d965f7bd189f6bd1cdc70436c41d
Reviewed-on: https://dawn-review.googlesource.com/c/dawn/+/88582
Commit-Queue: Loko Kung <lokokung@google.com>
Reviewed-by: Austin Eng <enga@chromium.org>
Kokoro: Kokoro <noreply+kokoro@google.com>
(cherry picked from commit 03ddfbb81fb4127ca37ea53e70fcb34fe851e24e)
---
 third_party/dawn/generator/dawn_version_generator.py | 13 ++++++++-----
 1 file changed, 8 insertions(+), 5 deletions(-)

diff --git a/third_party/dawn/generator/dawn_version_generator.py b/third_party/dawn/generator/dawn_version_generator.py
index 1907e88da..3c1927bee 100644
--- a/third_party/dawn/generator/dawn_version_generator.py
+++ b/third_party/dawn/generator/dawn_version_generator.py
@@ -23,11 +23,14 @@ def get_git():


 def get_gitHash(dawnDir):
-    result = subprocess.run([get_git(), 'rev-parse', 'HEAD'],
-                            stdout=subprocess.PIPE,
-                            cwd=dawnDir)
-    if result.returncode == 0:
-        return result.stdout.decode('utf-8').strip()
+    try:
+        result = subprocess.run([get_git(), "rev-parse", "HEAD"],
+                                stdout=subprocess.PIPE,
+                                cwd=dawnDir)
+        if result.returncode == 0:
+            return result.stdout.decode("utf-8").strip()
+    except Exception:
+        return ""
     # No hash was available (possibly) because the directory was not a git checkout. Dawn should
     # explicitly handle its absenece and disable features relying on the hash, i.e. caching.
     return ''
--
2.36.0
